test/testcondition.cpp

```diff
@@ -1718,6 +1719,91 @@ class TestCondition : public TestFixture {
         ASSERT_EQUALS("", errout.str());
     }
 
+    // operator precedence of = and && in condition
+    void clarifyAssignmentAndLogicalCondition() {
+        check("extern int foo();\n"
+              "extern bool bar();\n"
+              "void f() {\n"
+              "    if (x = foo() && bar()) {}\n" // don't simplify and verify this code
```

Comment:
I am skeptic about the comment. what do you want to say with that? this test case is not meant to test simplifications it's meant to test the checker.